### ISSUE TITLE
val_loss requirement

### ISSUE THREAD

**Issue by jakubMitura14** [speediedan/finetuning-scheduler#12]
hello thanks for publishing your work to the community !
As I see it is set to automatic that val_loss is monitored variable ; hence it is required for validation epoch to log the validation loss that will be named exactly like that "val_loss" I would suggest to add this info to docs or add monitor variable name to constructor , as it can cost couple hours of debugging to dig it out (as in my case) :)

**Comment by speediedan** [comment]
Thanks for your valuable feedback @jakubMitura14! In light of your experience I've just [pushed a commit](5d80dec98cfd5063c0097172183fd53b3b057d3f) that I think accordingly improves FTS usability.

#### Current State - (_as of v2.1.3_)

1. If one passes the desired monitor metric to `FTSEarlyStopping` as is done [in this test](https://github.com/speediedan/finetuning-scheduler/blob/ee0709875e60096b54d651f5d0d43965662a9317/tests/test_finetuning_scheduler_callback.py#L2575-L2577) or via a yaml config like [in this example](https://github.com/speediedan/finetuning-scheduler/blob/4607d09c9200166182355defc218f7a79e825354/src/fts_examples/stable/config/fts_explicit.yaml#L12-L14), the monitor metric used can be specified as desired since `FTSEarlyStopping` just uses the same "current"  metric logic used by Lightning, i.e., [current metric from the trainer](https://github.com/Lightning-AI/pytorch-lightning/blob/1a59097ab254e1cd85dc2cb88edfb6bc55087adc/src/lightning/pytorch/callbacks/early_stopping.py#L205-L206) is [evaluated by FTSEarlyStopping](https://github.com/speediedan/finetuning-scheduler/blob/ee0709875e60096b54d651f5d0d43965662a9317/src/finetuning_scheduler/fts_supporters.py#L279-L287)
2. While one can configure `FTSEarlyStopping` with an arbitrary monitor metric, as you note, if [no `FTSEarlyStopping` callback is provided](https://finetuning-scheduler.readthedocs.io/en/v2.1.3/index.html?highlight=val_loss#basic-usage:~:text=using%20default%20FTSEarlyStopping%20and%20FTSCheckpoint%20callbacks%20with%20monitor%3Dval_loss), a ``val_loss`` monitor metric is passed to the `FTSEarlyStopping` callback constructor (and `FTSCheckpoint` for that matter) as part of a default configuration . 
3. In cases where an early-stopping callback is provided that isn't fine-tuning schedule capable, (e.g. Lightning's `EarlyStopping`) the default `FTSEarlyStopping` callback configuration (including a ``val_loss``) metric is used and the relevant [variant of these warnings](https://github.com/speediedan/finetuning-scheduler/blob/ee0709875e60096b54d651f5d0d43965662a9317/src/finetuning_scheduler/fts_supporters.py#L1862-L1874) is provided.

Especially when providing a non-finetuning schedule capable callback like `EarlyStopping` with metric 'xyz' (scenario 3 above), I can see how the warnings I reference above might not be sufficiently clear about the default callback configuration replacing that metric with the default ``monitor='val_loss'``configuration. 

#### New State w/ Usability Improvements - (FTS >= v2.2.0) 

[The commit](5d80dec98cfd5063c0097172183fd53b3b057d3f) I've just pushed improves usability by:

1. Inspecting any base `EarlyStopping` or `ModelCheckpoint` configuration passed to FTS by the user and using that configuration to instantiate the required FTS callback dependencies (i.e., `FTSEarlyStopping` or `FTSCheckpoint`). 
2. Adding a [note in the documentation](https://finetuning-scheduler.readthedocs.io/en/latest/#:~:text=If%20not%20provided%2C%20FTS%20will%20instantiate%20its%20callback%20dependencies) indicating more clearly the behavior of FTS when no monitor metric configuration is provided.

Hope you've found FTS of some utility once working through the issue you encountered. 

Thanks again for your contribution, you've helped improve FTS for everyone! Feel free to reach out anytime if you have other issues or want to share more about your use case. Best of luck with your work!

**Comment by jakubMitura14** [comment]
Thanks for fast and professional response !
